sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/Main.java

```diff
@@ -146,13 +164,77 @@ private void appendPerf(
   private static final String LINE =
       "==========================================================================================";
 
-  /**
-   * Print summary  of {@code actual} vs (if non-null) {@code baseline}.
-   */
+  /** Send {@code nexmarkPerf} to BigQuery. */
+  @VisibleForTesting
+  static void writeQueryPerftoBigQuery(
```

Comment:
Can we perform this write using a Beam pipeline that uses DirectRunner and BigQuery sink instead of using BigQueryService class ? I think you should use either that or use a BigQuery client library directly instead of using the intermediate "BigQueryService" class in Beam which is not intended to be a public utility.

Reply:
For your 2 questions:
1. No, here we output to BigQuery only the response time of the pipeline so it needs to be finished. 
2. What you suggest (bigquery client) is what I did at first and I asked you how I can unit test it and you answered that I should use FakeBigQueryServices. But to use FakeBigQueryServices I needed to refactor everything to use FakeBigQueryServices's super class (BigQueryServices)

Reply:
Using FakeBigQueryService for testing is fine, but seems like here you are using the BigQueryService class (which is supposed to be an implementation detail of Beam BigQuery connector) to publish results to real BigQuery, no ? (apologies if I misunderstood) I was suggesting to use a second Beam pipeline that uses DirectRunner (not the original pipeline) to publish results to BigQuery instead. I don't think  we should expand the public interface of BigQuery connector for this task.

Reply:
True, I'm using BigQueryService class. If you don't want it exposed, I can use BigQueryIO in a second pipeline indeed. I can create this pipeline once the run pipeline is finished and run it over of PCollection<KV<Configuration, NexmarkPerf>> and write this Pcollection to BQ using BigQueryIO.

Reply:
Using any runner (for the second pipeline) is fine. But can you clarify why you cannot use DirectRunner here ? Using DirectRunner will be the easiest option if you want to push a small amount of data to BigQuery. You'll have to create a new PipelineOptions object for this.